Add bdo example

ericwbailey opened the pull request:
This PR addresses the issue outlined in https://github.com/w3c/html/issues/1254.

It adds an example of the `<bdo>` element. If the example feels too meta, I'm happy to revise.

ericwbailey committed:
Add bdo example

https://github.com/w3c/html/issues/1254

ericwbailey committed:
Re-phrased example

edent commented:
A good start! I think it would be helpful to show exactly what result is it is used incorrectly.

chaals commented:
Marked as non substantive for IPR from ash-nazg.

ericwbailey committed:
Clarified examples

ericwbailey commented:
Thank you for the feedback, @edent! I just pushed up an update that expands on the example a little bit more.

edent commented on sections/semantics-textlevel.include:
```diff
@@ -2957,11 +2957,28 @@
+  <div class="example">
+      This example shows how the <{bdo}> element can be used to alter directionality formatting:
+
+      <bdi><bdo dir="rtl">English is a left-to-right language.</bdo></bdi>
```
I think this demonstration should come after the example code.

edent commented on sections/semantics-textlevel.include:
```diff
@@ -2957,11 +2957,28 @@
+      If character encoding isn't appropriately defined, the characters of <abbr>RTL</abbr> languages
+      may appear in reverse order. For example:
+
+      <bdi><bdo dir="ltr">עֶזרָה</bdo></bdi>
```
Can you add in some `<xmp>` example code so that readers can see how this was achieved?

edent replied:
Specifically because people who can't read Hebrew won't be able to tell what's wrong.

chaals replied:
Actually, can you explicitly say what is going on with the example? E.g. hebrew characters normally read right to left, so ...

(You might want to use `ruby` on the characters, or something that indicates what happened).

ericwbailey replied:
I was under the impression that `ruby` was reserved for kanji?  Otherwise, does

>In this example, Hebrew characters have been reversed from their normal right to left order.

sound like an acceptable explanation for the `xmp` example code?

chaals replied:
no,, you can use `ruby` wherever it makes sense. But yes, the text makes sense - thanks.

r12a replied:
(fwiw, actually ruby doesn't support all you need for a general annotation mechanism, and is developed with the particular needs of CJKM in mind, so i don't usually recommend using it for general cases unless one proceeds with caution, and one avoids a suggesting to uninformed readers  that ruby is an adequate general solution.)

ericwbailey committed:
Revisions based on feedback

https://github.com/w3c/html/pull/1267#discussion_r171803531
https://github.com/w3c/html/pull/1267#discussion_r171803656

ericwbailey commented:
Pushed up changes based on your comments. Thanks for the feedback and patience as I stumble my way through this.

chaals commented on sections/semantics-textlevel.include:
```diff
@@ -2957,11 +2957,32 @@
+  </div>
+
+  <div class="example">
+      If character encoding isn't appropriately defined, the characters of <abbr>RTL</abbr> languages
```
appropriately defined should link somewhere, no? But see also #1273 where we might change this...

ericwbailey replied:
Ah neat, I wasn't aware the spec recommended UTF-8!

chaals commented:
(More nagging... :) ). Thank you for taking this on. I have also asked the i18n group to take a look, as they may have even more comments... or helpful suggestions on improving this bit.

ericwbailey committed:
Better described bdo example, linked to encoding

ericwbailey commented:
Just pushed up changes for explaining the RTL reversal example, as well as linking to language on encoding. I'll lurk in https://github.com/w3c/html/pull/1273 as well.

chaals commented:
I think we could still improve on this, but it's better than where we are now so let's move forward.

aphillips commented:
I don't think the bottom example makes sense?? It says:

> If character encoding isn’t appropriately defined, the characters of RTL languages may appear in reverse order.

The point of `bdo` is that it overrides the Unicode bidi algorithm. In this case, it says to show the characters in ltr order (effectively the same as the English example just above). This has absolutely nothing to do with the character encoding (except, maybe, possibly when dealing with legacy visual encodings of RTL languages--but even then the characters are already Unicode by the time they are in the DOM). I think this text needs more help...

chaals commented:
@aphillips

> I think this text needs more help...

Indeed it does, and we would be grateful for any. 

(I may be able to work on this on Sunday, but certainly not earlier, and I think we have demonstrated that I am not the expert we are looking for so I will be starting with a bit more study and more careful testing.)

r12a commented:
@chaals @ericwbailey during this week's i18n call I was given an action to suggest some text for you. How soon do you need it?

chaals commented:
Tomorrow would be great. Before the end of the month would be nice. It would be bad if we did not fix this before the closing date of the milestone - April 6.
